### DISCUSSION TITLE
How to enforce a set of global facet filters that are always applied?

### DISCUSSION THREAD

**Discussion by grahammcculloch** [algolia/instantsearch#5290] [Q&A] [answered]
My use-case is this: in 'normal' InstantSearch use, my 'states' facet widget allows selection of any combination of all 50 US states. But, based on user-specific configuration, I want to limit the 'state' options to just those that are listed in that user's configuration (e.g. this user can only see items in these three states [ 'California', 'Nevada', 'Arizona' ] and can select one or more of those states from the 'states' facet widget). The other facet options (e.g. 'brand') should reflect the limited set of states (e.g. only list brands that relate to those three states).

Ideally this 'global' filter should be applied when querying Algolia, but should not appear in the searchState (and should not set those globally-filtered states as selected in the 'states' facet widget).

Can anyone suggest a neat way to implement something like this using react-instantsearch?

**Comment by Haroenv** [comment, marked as answer]
A user filter that shouldn't be visible is best done via api key restrictions. You can follow this guide: https://www.algolia.com/doc/api-reference/api-methods/generate-secured-api-key/#method-param-filters

https://www.algolia.com/doc/guides/solutions/ecommerce/b2b-catalog-management/tutorials/personalized-catalogs/#implementing-secured-api-keys

Hope that's clear!

**Reply by grahammcculloch** [reply]
Ah - fantastic, thanks for the links! This looks like exactly what we need.

One small follow-up question... the examples show just one filter being specified (`filters: '_tags:user_42'`). If I want to specify something like this:
```
FACET1 = A or B or C
AND
FACET2 = X or Y or Z
```
what would that look like in the `filters` stringified format?

**Reply by Haroenv** [reply]
It would look like `(facet1:A OR facet1:B OR facet1:C) AND (facet2:X OR facet2:Y OR facet2:Y)` @grahammcculloch, there's a syntax validator here: https://www.algolia.com/doc/api-reference/api-parameters/filters/#filters-syntax-validator

**Reply by grahammcculloch** [reply]
@Haroenv thanks!
